koku/api/settings/tag_management.py

```diff
@@ -0,0 +1,222 @@
+from reporting.models import OCPEnabledTagKeys
+
+
+obtainTagKeysProvidersParams = {
+    "openshift": {
+        "provider": Provider.PROVIDER_OCP,
+        "title": "OpenShift labels",
+        "leftLabel": "Available labels",
+        "rightLabel": "Labels for reporting",
+        "tag_view": OCPTagView,
+        "query_handler": OCPTagQueryHandler,
+        "enabled_tag_keys": OCPEnabledTagKeys,
+    },
+    "aws": {
+        "provider": Provider.PROVIDER_AWS,
+        "title": "Amazon Web Services tags",
+        "leftLabel": "Available tags",
+        "rightLabel": "Tags for reporting",
+        "tag_view": AWSTagView,
+        "query_handler": AWSTagQueryHandler,
+        "enabled_tag_keys": AWSEnabledTagKeys,
+    },
+    "azure": {
+        "provider": Provider.PROVIDER_AZURE,
+        "title": "Azure tags",
+        "leftLabel": "Available tags",
+        "rightLabel": "Tags for reporting",
+        "tag_view": AzureTagView,
+        "query_handler": AzureTagQueryHandler,
+        "enabled_tag_keys": AzureEnabledTagKeys,
+    },
+}
+
+
+class TagManagementSettings:
+    """Class for generating OpenShift settings."""
+
+    def __init__(self, request):
+        """Initialize settings object with incoming request."""
+        self.request = request
+        self.factory = RequestFactory()
+        self.schema = request.user.customer.schema_name
+
+    def _get_tag_management_prefix(self, providerName):
+        return f"{SETTINGS_PREFIX}.tag-management.{providerName}"
+
+    def _obtain_tag_keys(self, tag_view_kls, tag_handler_kls, tag_keys_kls):
+        """
+        Collect the available tag keys for the customer.
+
+        Returns:
+            (List) - List of available tag keys objects
+            (List) - List of enabled tag keys strings
+        """
+        url = (
+            "?filter[time_scope_units]=month&filter[time_scope_value]=-1"
+            "&filter[resolution]=monthly&key_only=True&filter[enabled]=False"
+        )
+        tag_request = self.factory.get(url)
+        tag_request.user = self.request.user
+        query_params = QueryParameters(tag_request, tag_view_kls)
+        handler = tag_handler_kls(query_params)
+        query_output = handler.execute_query()
+        avail_data = query_output.get("data")
+        all_tags_set = set(avail_data)
+        enabled = []
+        with schema_context(self.schema):
+            enabled_tags = tag_keys_kls.objects.all()
+            enabled = [enabled_tag.key for enabled_tag in enabled_tags]
+            all_tags_set.update(enabled)
+
+        return all_tags_set, enabled
+
+    def _build_tag_key(self):
+        """
+        Generate tag_key tab component
+
+        Returns:
+            (Dict) - Tab Item
+        """
+        tag_key_text_name = f"{self._get_tag_management_prefix('openshift')}.form-text"
+        tag_key_text_context = (
+            "Enable your OpenShift label names or Amazon Web Services or Azure tag keys to be used for report"
```

Comment:
Working with @nlcwong on revising this to reflect the fact that changes in the UI are made on next ingest.  If we remove all mentions of source types it would be less wordy and we wouldn't have to have DEVELOPMENT versions of the strings (for the interim where we don't have AWS or Azure)

Reply:
Can you update this `tag_key_text_context` string to be:

`Enable your data source labels to be used as tag keys for report grouping and filtering. Changes will be reflected within 24 hours.`